homeassistant/components/samsungtv/config_flow.py

```diff
@@ -408,7 +408,7 @@ def _async_update_existing_matching_entry(
             )
         return entry
 
-    async def _async_start_discovery_with_mac_address(self) -> None:
+    def _start_discovery_with_mac_address(self) -> None:
```

Comment:
Prefixing the name with `async_` means the function should run in the event loop. It doesn't mean that it's a coroutine function.

https://developers.home-assistant.io/docs/asyncio_categorizing_functions